src/components/media-loader.js

```diff
@@ -102,27 +98,56 @@ AFRAME.registerComponent("media-loader", {
     const mesh = useFancyLoader
       ? loadingObject.scene.clone()
       : new THREE.Mesh(new THREE.BoxGeometry(), new THREE.MeshBasicMaterial());
+    this.el.setObject3D("mesh", mesh);
+    this.updateScale(true);
     if (useFancyLoader) {
       this.loaderMixer = new THREE.AnimationMixer(mesh);
+
       this.loadingClip = this.loaderMixer.clipAction(loadingObject.animations[0]);
+      this.loadingScaleClip = this.loaderMixer.clipAction(
+        new THREE.AnimationClip(null, 1000, [
+          new THREE.VectorKeyframeTrack(".scale", [0, 0.2], [0, 0, 0, mesh.scale.x, mesh.scale.y, mesh.scale.z])
+        ])
+      );
+      setTimeout(() => {
+        if (!this.loaderMixer) return; // Animation/loader was stopped early
+        this.el.setAttribute("ammo-shape__loader", { type: SHAPE.BOX });
+      }, 200);
+
       this.loadingClip.play();
+      this.loadingScaleClip.play();
     }
-    this.el.setObject3D("mesh", mesh);
-    this.setShapeAndScale(true, SHAPE.BOX, "loader");
     delete this.showLoaderTimeout;
   },
 
   clearLoadingTimeout() {
     clearTimeout(this.showLoaderTimeout);
     if (this.loaderMixer) {
       this.loadingClip.stop();
+      this.loadingScaleClip.stop();
       delete this.loaderMixer;
+      delete this.loadingScaleClip;
+      delete this.loadingClip;
     }
     delete this.showLoaderTimeout;
+    this.removeShape("loader");
+  },
 
-    this.waitForMediaScaleReady(() => {
-      this.removeShape("loader");
-    });
+  updateShape(shapeType) {
```

Comment:
Would prefer this to take an `options` or `props` object like so:

```
    const options = {type: SHAPE.BOX,
        halfExtents: { x: 0.5, y: 0.5, z: 0.02 },
        margin: 0.1,
        fit: FIT.MANUAL
    }
    updateShape(options);
...
   updateShape(shapeOptions) {
        if (this.el.getAttribute("ammo-shape")) {
    });
      this.el.removeAttribute("ammo-shape");
    }
      this.el.setAttribute("ammo-shape", shapeOptions);
    }
```
Obviously also do the same where this is being called in media-views